### ISSUE TITLE
Xgboost==0.90 Issue with docker 

### ISSUE THREAD

**Issue by Roopesh-Bharatwaj-K-R** [dmlc/xgboost#8126]
Details:

Operating System : Mac Os
Python Version : 3.8

Problem :

in My python Virtual Conda environment, pip install xgboost ==0.90 works fine and my model gives me everything correct without any dependency issue 

when i tried to create docker image, In docker file i have given pip install xgboost ==0.90, but this specific version throws me error and not allowing me to create the docker image.

     => ERROR [8/8] RUN pip install xgboost ==0.90                             0.4s
     ------
     > [8/8] RUN pip install xgboost ==0.90:
     #12 0.358 ERROR: Invalid requirement: '==0.90'
     ------
     executor failed running [/bin/sh -c pip install xgboost ==0.90]: exit code: 1

When in docker file, if i remove the specific version '==0.90', docker installation is fine but the model does not run and throws the below error

    File "/usr/local/lib/python3.8/site-packages/xgboost/core.py", line 203, in _check_call
        raise XGBoostError(py_str(_LIB.XGBGetLastError()))
    xgboost.core.XGBoostError: [11:42:25] /workspace/src/learner.cc:1059: Check failed: header == serialisation_header_: 
      If you are loading a serialized model (like pickle in Python, RDS in R) generated by
      older XGBoost, please export the model by calling `Booster.save_model` from that version
      first, then load it back in current version. See:

    https://xgboost.readthedocs.io/en/latest/tutorials/saving_model.html

Please help me to solve this issue, and to install in docker file 'xgboost==0.90'
Thanks in Advance.

**Comment by trivialfis** [comment]
Can you remove that extra space before the ==?

**Comment by trivialfis** [comment]
```
pip install xgboost ==0.90
ERROR: Invalid requirement: '==0.90'
```
I tried running pip this way and it matches your error.  Simply removing the space should fix it.

**Comment by Roopesh-Bharatwaj-K-R** [comment]
> Can you remove that extra space before the ==?

Hi, First Thanks for the Response, I removed extra space but the issue when building docker is the same,

 
      => ERROR [8/8] RUN pip install xgboost==0.90                                                                                                                                                          1.1s 
------                                                                                                                                                                                                      
 > [8/8] RUN pip install xgboost==0.90:                                                                                                                                                                     
#13 0.433 Collecting xgboost==0.90                                                                                                                                                                          
#13 0.517   Downloading xgboost-0.90.tar.gz (676 kB)                                                                                                                                                        
#13 0.807      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 676.7/676.7 kB 2.4 MB/s eta 0:00:00
#13 0.905   Preparing metadata (setup.py): started
#13 0.987   Preparing metadata (setup.py): finished with status 'error'
#13 0.990   error: subprocess-exited-with-error
#13 0.990   
#13 0.990   × python setup.py egg_info did not run successfully.
#13 0.990   │ exit code: 1
#13 0.990   ╰─> [21 lines of output]
#13 0.990       + pwd
#13 0.990       + oldpath=/tmp/pip-install-uyxzxu8s/xgboost_f213fe1c74094083a5d4695282eab66f
#13 0.990       + cd ./xgboost/
#13 0.990       + echo
#13 0.990       + grep -q darwin
#13 0.990       + LIB_XGBOOST=libxgboost.so
#13 0.990       + make clean
#13 0.990       ./xgboost/build-python.sh: 40: make: not found
#13 0.990       Traceback (most recent call last):
#13 0.990         File "<string>", line 2, in <module>
#13 0.990         File "<pip-setuptools-caller>", line 34, in <module>
#13 0.990         File "/tmp/pip-install-uyxzxu8s/xgboost_f213fe1c74094083a5d4695282eab66f/setup.py", line 42, in <module>
#13 0.990           LIB_PATH = libpath['find_lib_path']()
#13 0.990         File "/tmp/pip-install-uyxzxu8s/xgboost_f213fe1c74094083a5d4695282eab66f/xgboost/libpath.py", line 45, in find_lib_path
#13 0.990           raise XGBoostLibraryNotFound(
#13 0.990       XGBoostLibraryNotFound: Cannot find XGBoost Library in the candidate path, did you install compilers and run build.sh in root path?
#13 0.990       List of candidates:
#13 0.990       /tmp/pip-install-uyxzxu8s/xgboost_f213fe1c74094083a5d4695282eab66f/xgboost/libxgboost.so
#13 0.990       /tmp/pip-install-uyxzxu8s/xgboost_f213fe1c74094083a5d4695282eab66f/xgboost/../../lib/libxgboost.so
#13 0.990       /tmp/pip-install-uyxzxu8s/xgboost_f213fe1c74094083a5d4695282eab66f/xgboost/./lib/libxgboost.so
#13 0.990       /usr/local/xgboost/libxgboost.so
#13 0.990       [end of output]
#13 0.990   
#13 0.990   note: This error originates from a subprocess, and is likely not a problem with pip.
#13 0.991 error: metadata-generation-failed
#13 0.991 
#13 0.991 × Encountered error while generating package metadata.
#13 0.991 ╰─> See above for output.
#13 0.991 
#13 0.991 note: This is an issue with the package mentioned above, not pip.
#13 0.991 hint: See above for details.
------
       executor failed running [/bin/sh -c pip install xgboost==0.90]: exit code: 1




This is the only Issue for creating Docker File and installing this specific Version, the same version i'm using in Conda Environment which works fine.

I tried builded on the specific file location of my Project.

Kindly help me back, how to overcome this issue and Build Docker file on that specific 'Xgboost==0.90' version

**Comment by hcho3** [comment]
It seems that the binary package wasn't yet available for 0.90. That's why pip is attempting to install from the source tarball xgboost-0.90.tar.gz.

If you have a special need for 0.90 version, please use Conda.
